jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/internal/HttpChannelState.java

```diff
@@ -222,6 +226,34 @@ public ByteBufferPool getByteBufferPool()
         return getConnectionMetaData().getConnector().getByteBufferPool();
     }
 
+    @Override
+    public ByteBufferPool.Sized getInputByteBufferPool()
+    {
+        return _inputByteBufferPool.updateAndGet(this::checkInputPool);
```

Comment:
It looks like this patter could be extracted in a specific helper class that would take an init lambda in its constructor, maybe `LazyAtomicReference`?

Reply:
Nah, that's not a great idea in practice.